internal/topic/core_test.go

```diff
@@ -547,3 +547,64 @@ func TestCore_Get(t *testing.T) {
 		})
 	}
 }
+
+func TestCore_SetupTopicRetentionConfigs(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	dlqTopic := getDLQDummyTopicModel()
+
+	mockTopicRepo := topicrepomock.NewMockIRepo(ctrl)
+	mockAdmin := messagebrokermock.NewMockBroker(ctrl)
+
+	tests := []struct {
+		name            string
+		topics          []common.IModel
+		existingConfigs map[string]map[string]string
+		expected        []string
+		listErr         error
+		getAdminErr     error
+		expectedErr     error
+	}{
+		{
+			name:     "Alter retention configs without errors",
+			topics:   []common.IModel{dlqTopic},
+			expected: []string{dlqTopic.Name},
```

Comment:
Can you add a test case with 4 topics and 1 dlq topic. We need to assert the the AlterTopicConfigs is called only with dlq topic with retention changed.